### DISCUSSION TITLE
Trait HasRelationships::addRelation should accept a forth parameter to override existent relation

### DISCUSSION THREAD

**Discussion by goldmont** [wintercms/winter#1139] [Ideas]
Hi,

The current behaviour of HasRelationships::addRelation is to fail when the developer tries to override an existing relation via the `add*Relation()` methods. This is due to [this](https://github.com/wintercms/storm/blob/develop/src/Database/Concerns/HasRelationships.php#L816) line. I expect to have the ability to override an existing relation because currently I have to use the PHP native way.

**Example**
`\Backend\Models\User` model:
I would like to override the value of the `avatar` relation in order to use my custom `File::class` implementation.
```php
public $attachOne = [
    'avatar' => \System\Models\File::class
];
```

The following approach doesn't work:
```php
\Backend\Models\User::extend(function ($model) {
	$model->addAttachOneRelation('avatar', [
		File::class
	]);
});
```
It fails with the following error:

> Cannot add the "avatar" relation to Backend\Models\User, it conflicts with an existing relation, attribute, or property. in /var/www/vendor/winter/storm/src/Database/Concerns/HasRelationships.php:817

As a workaround, I can do this:
```php
\Backend\Models\User::extend(function ($model) {
	$model->attachOne['avatar'] = [
		File::class
	];
});
```

---

It would be nice to do something like this:
```php
\Backend\Models\User::extend(function ($model) {
	$model->addAttachOneRelation('avatar', [
		File::class
	], override: true);
});
```

What do you think about it? Thank you.

**Comment by mjauvin** [comment]
Please submit a PR, I also think this is a welcome improvement.

**Comment by goldmont** [comment]
Sure. `override` is good to you as parameter name?

**Comment by mjauvin** [comment]
Yes, if not the other maintainers will suggest an alternative.

**Reply by goldmont** [reply]
I'll wait till tomorrow.

**Reply by mjauvin** [reply]
No need, go ahead with the PR and that can be suggested there
